### DISCUSSION TITLE
SAM E70(B) Xplained Ultra

### DISCUSSION THREAD

**Discussion by bathm** [zephyrproject-rtos/zephyr#38099] [Q&A]
Hi,
I want to run Zephyr on a SAM E70B Xplained ULTRA (XULT) eval board. In the board section there is only the SAM E70 XPLAINED (XPLD) board. I edited the .dtsi to adapt it to the schematic of the XULT (USER LED on PA05). 
When I build and flash the sample samples/basis/button the firmware runs nice on the XPLD board. When I flash it to the XULT board there are only hieroglyphs at the UART output.

Here is the minicom output:

![SAME70_XPLD](https://user-images.githubusercontent.com/89775938/131319985-1a9f8080-d7ba-454d-a77d-78bcee16ef06.png)
![SAME70_XULT](https://user-images.githubusercontent.com/89775938/131319995-d35b2a24-c2b2-4d6e-87b5-a4636350e2fe.png)

Any ideas? Thanks!

**Comment by nandojve** [comment]
Hi @bathm ,

Looking at schematics, both boards (E70_XPLD and E70_XULT) are using same UART-1 for console and are using same 12MHz crystals.
In this case, it should work. It'll be necessary investigate more to catch the problem and you need provide more information.

Note: SAM's boards with chip revision B have an explicit **b**, like sam_e70**b**_xplained, to select chip revision B headers files.

 - Are you using below command to build your board to compile code with ?
`west build -b sam_e70b_xplained samples/basic/button`
 - What Zephyr version/revision are you using? #code

**Reply by bathm** [reply]
Hi @nandojve,
thank you for your reply.
To exclude the possibility of any hardware issues I've also tried another E70_XULT from a fellow student with the same result.

- Yes, I'm using chip revision B on both boards and I'm using the command `west build -b sam_e70b_xplained samples/basic/button`
- I'm using Zephyr v2.6.0-rc1

I'll compare both schematics again and try to find the difference between them.

**Comment by nandojve** [comment]
Hi @bathm ,

The evolution of Zephyr from one version to another is substantial. The current upstream/main is in stabilization phase for 2.7.0 ( LTS ), scheduled by end Sep/21.

I strong recommend you to use Zephyr main till LTS release.

 - source: upstream/main (2.7.0 - LTS)
 - SDK 0.13.0
